ui/page/staking/auto_ticket_modal.go

```diff
@@ -0,0 +1,230 @@
+package staking
+
+import (
+	"fmt"
+	"strconv"
+
+	"gioui.org/layout"
+	"gioui.org/text"
+	"gioui.org/widget"
+
+	"github.com/planetdecred/dcrlibwallet"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/load"
+	"github.com/planetdecred/godcr/ui/page/components"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+const ticketBuyerModalID = "staking_modal"
+
+type ticketBuyerModal struct {
+	*load.Load
+
+	settingsSaved func()
+	onCancel      func()
+
+	modal           decredmaterial.Modal
+	cancel          decredmaterial.Button
+	saveSettingsBtn decredmaterial.Button
+
+	balToMaintainEditor decredmaterial.Editor
+
+	accountSelector *components.AccountSelector
+	vspSelector     *vspSelector
+}
+
+func newTicketBuyerModal(l *load.Load) *ticketBuyerModal {
+	tb := &ticketBuyerModal{
+		Load: l,
+
+		cancel:          l.Theme.OutlineButton("Cancel"),
+		saveSettingsBtn: l.Theme.Button("Save"),
+		modal:           *l.Theme.ModalFloatTitle(),
+		vspSelector:     newVSPSelector(l).title("Select a vsp"),
+	}
+
+	tb.balToMaintainEditor = l.Theme.Editor(new(widget.Editor), "Balance to maintain (DCR)")
+	tb.balToMaintainEditor.Editor.SingleLine = true
+
+	tb.saveSettingsBtn.SetEnabled(false)
+
+	return tb
+}
+
+func (tb *ticketBuyerModal) OnSettingsSaved(settingsSaved func()) *ticketBuyerModal {
+	tb.settingsSaved = settingsSaved
+	return tb
+}
+
+func (tb *ticketBuyerModal) OnCancel(cancel func()) *ticketBuyerModal {
+	tb.onCancel = cancel
+	return tb
+}
+
+func (tb *ticketBuyerModal) selectFirstWalletAccount() {
+	err := tb.accountSelector.SelectFirstWalletValidAccount()
+	if err != nil {
+		tb.Toast.NotifyError(err.Error())
+	}
+}
+
+func (tb *ticketBuyerModal) OnResume() {
+	tb.initializeAccountSelector()
+	if len(tb.WL.MultiWallet.VspList) == 0 {
+		go tb.WL.MultiWallet.GetVSPList(tb.WL.Wallet.Net)
+	}
+
+	// loop through all available wallets and select the one with ticket buyer config.
+	// if non, set the selected wallet to the first
```

Comment:
issue with that approach is that it for any new wallet, it gets same old stored value, which  causes some unwanted error logs. 

This approach however, brings us closer to extending the ticket buyer to multi wallets instead of 1 as the configs are stored on a per wallet bases.